### DISCUSSION TITLE
HDMI PC Mode API help

### DISCUSSION THREAD

**Discussion by alimbada** [JPersson77/LGTVCompanion#171] [General]
Hi,

I noticed you have a command that allows changing the HDMI input to PC mode. Are you able to point me to the LG API which is called when this command is invoked? I would like to add support for it to [klattimer/LGWebOSRemote](https://github.com/klattimer/LGWebOSRemote) as this feature is missing from it.

Thanks,

Ammaar

**Comment by dechamps** [comment]
You can look at the [relevant LGTVCompanion code](https://github.com/JPersson77/LGTVCompanion/blob/83073254c70344adb692f335fcff47db902ccb53/Common/lg_api_commands.h#L380-L403) or the [equivalent code in bscpylgtv](https://github.com/chros73/bscpylgtv/blob/8267636a991bacbb581bca57b6b461adf1df05a2/bscpylgtv/webos_client.py#L1465) for inspiration.

**Comment by JPersson77** [comment]
Hi @alimbada 

Additionally, there is this command:

`-set_input_type [hdmi_input] [icon] [label] `

You need to dispatch a "luna" event with the following payload:

"id" = [hdmi_input]
"icon" = [icon].png
"label" = [label]

to this endpoint:

luna://com.webos.service.eim/setDeviceInfo

**Comment by JPersson77** [comment]
To add to @dechamps reply. This method is also a "luna" message. But to the endpoint  luna://com.webos.settingsservice/setSystemSettings with "category" = "other"
